test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -390,11 +390,11 @@ BOOST_AUTO_TEST_CASE(unsatisfied_version)
 BOOST_AUTO_TEST_CASE(returndatasize_as_variable)
 {
 	char const* text = R"(
-		contract c { function f() public { uint returndatasize; assembly { returndatasize }}}
+		contract C { function f() public pure { uint returndatasize; assembly { returndatasize() }}}
 	)";
 	vector<pair<Error::Type, std::string>> expectations(vector<pair<Error::Type, std::string>>{
 		{Error::Type::Warning, "Variable is shadowed in inline assembly by an instruction of the same name"},
-		{Error::Type::Warning, "The use of non-functional instructions is deprecated."},
+		{Error::Type::SyntaxError, "Top-level expressions are not supposed to return values"},
 		{Error::Type::DeclarationError, "Unbalanced stack"}
```

Comment:
Probably `pop(returndatasize()`` would be better to avoid these two errors.

Reply:
I agree, however, since the error was there before already I didn't feel quite sure of whether to `pop` it or not. Thanks for clarifying.